confmap/resolver.go

```diff
@@ -98,32 +87,15 @@ func NewResolver(set ResolverSettings) (*Resolver, error) {
 		set.ConverterSettings.Logger = zap.NewNop()
 	}
 
-	var providers map[string]Provider
-	var converters []Converter
-
-	if len(set.Providers) != 0 {
-		if len(set.ProviderFactories) != 0 {
-			return nil, errors.New("only one of ResolverSettings.Providers and ResolverSettings.ProviderFactories can be used")
-		}
-		providers = set.Providers
-	} else {
-		providers = make(map[string]Provider, len(set.ProviderFactories))
-		for _, factory := range set.ProviderFactories {
-			provider := factory.Create(set.ProviderSettings)
-			providers[provider.Scheme()] = provider
-		}
+	providers := make(map[string]Provider, len(set.ProviderFactories))
+	for _, factory := range set.ProviderFactories {
+		provider := factory.Create(set.ProviderSettings)
+		providers[provider.Scheme()] = provider
 	}
 
-	if len(set.Converters) != 0 {
-		if len(set.ConverterFactories) != 0 {
-			return nil, errors.New("only one of ResolverSettings.Converters and ResolverSettings.ConverterFactories can be used")
-		}
-		converters = set.Converters
-	} else {
-		converters = make([]Converter, len(set.ConverterFactories))
-		for i, factory := range set.ConverterFactories {
-			converters[i] = factory.Create(set.ConverterSettings)
-		}
+	converters := make([]Converter, len(set.ConverterFactories))
+	for i, factory := range set.ConverterFactories {
+		converters[i] = factory.Create(set.ConverterSettings)
```

Comment:
Separate PR/issue. It would be great to have all public APIs like `Create` always accept `context.Context` as the first argument.